lighthouse-core/lib/third-party-web.js

```diff
@@ -8,19 +8,22 @@
 const thirdPartyWeb = require('third-party-web/httparchive-nostats-subset');
 
 /** @typedef {import("third-party-web").IEntity} ThirdPartyEntity */
+/** @typedef {import("third-party-web").IProduct} ThirdPartyProduct */
 
 /**
- * `third-party-web` throws when the passed in string doesn't appear to have any domain whatsoever.
- * We pass in some not-so-url-like things, so make the dependent-code simpler by making this call safe.
  * @param {string} url
  * @return {ThirdPartyEntity|undefined}
  */
 function getEntity(url) {
-  try {
-    return thirdPartyWeb.getEntity(url);
-  } catch (_) {
-    return undefined;
-  }
+  return thirdPartyWeb.getEntity(url);
+}
+
+/**
+ * @param {string} url
+ * @return {ThirdPartyProduct|undefined}
+ */
+function getProduct(url) {
```

Comment:
We make use of the `isFirstParty` function in `third-party-facades.js`. I would rather not reimplement that or use two different `thirdPartyWeb` modules so I prefer leaving this TBH.

Reply:
Also non-trivial to remember to fetch the *same* no stats bundle subset to avoid duplication so I'd also prefer to have all 3p usages come through here 👍